javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -481,13 +479,13 @@ default int length() {
     @Override
     default <U> Seq<U> scanLeft(U zero, BiFunction<? super U, ? super Tuple2<K, V>, ? extends U> operation) {
         Objects.requireNonNull(operation, "operation is null");
-        return Collections.scanLeft(this, zero, operation, List.empty(), List::prepend, List::reverse);
+        return Collections.scanLeft(this, zero, operation, Stream.empty(), Stream::append, Function.identity());
     }
 
     @Override
     default <U> Seq<U> scanRight(U zero, BiFunction<? super Tuple2<K, V>, ? super U, ? extends U> operation) {
         Objects.requireNonNull(operation, "operation is null");
-        return Collections.scanRight(this, zero, operation, List.empty(), List::prepend, Function.identity());
+        return Collections.scanRight(this, zero, operation, Stream.empty(), Stream::prepend, Function.identity());
```

Comment:
Stream makes most sense when doing it lazily (for maybe infinite amount of elements). The Stream::prepend currently evaluates the whole Seq eagerly. Therefore please use Vector here instead of Stream.

The benefit of Vector: It is also great for random access (approx. O(1)), Stream only has O(n) for random access